### DISCUSSION TITLE
Windows10 can't read HTTP_PROXY environment variable

### DISCUSSION THREAD

**Discussion by Checkpoints** [denoland/deno#9637] [Q&A] [answered]
I have asked the same question at Stackflow. Not sure if it's my code wrong, or it's a Windows related bug.

I set the Proxy url like
Deno.env.set('HTTP_PROXY', 'https://138.68.60.8:3128/');
Deno.env.set('HTTPS_PROXY', 'https://138.68.60.8:3128/');

In face the manual doesn't say how shall I set the environment variables. These are googled from somewhere.

yet the fetch() still fetches without the proxy. Instead, if I set up a proxy in the Windows system setting, the fetch() does use the system set proxy. Which doesn't solve the problem though as I am trying to read the proxy list from an API.

**Comment by lucacasonato** [comment, marked as answer]
You have to see these environment variables before you start Deno, not from within the runtime.

**Reply by Checkpoints** [reply]
Thank you a lot. May I ask what's the proper way to setup the environment variables in Windows? apparently my way doesn't work and I can't find any instruction about it...

**Reply by kitsonk** [reply]
https://www.google.com/search?q=set+environment+variables+in+windows+10

**Reply by Checkpoints** [reply]
Sure, but do we have any other ways to load proxies in Deno other than reading the hardcoded windows environment variables?

Since we may want to set proxies in code or say in run time. For example, liberating a proxy pool. In Node js we can do it I believe.
